syntax = "proto2";

message Font {
  optional string name = 1;
  optional string style = 2;
  optional float weight = 3;
  optional string filename = 4;
  optional string post_script_name = 5;
  optional string full_name = 6;
  optional string copyright = 7;
}

message Axes {
  optional string tag = 1;
  optional float min_value = 2;
  optional float max_value = 3;
}

message Source {
  optional string repository_url = 1;
  optional string commit = 2;
  optional string archive_url = 3;
}

message RegistryDefaultOverrides {
  optional string key = 1;
  optional float value = 2;
}

message FontInfo {
  optional string name = 1;
  optional string designer = 2;
  optional string license = 3;
  repeated string category = 4;
  optional string date_added = 5;
  repeated Font fonts = 6;
  repeated string subsets = 7;
  repeated Axes axes = 8;
  optional Source source = 9;
  repeated RegistryDefaultOverrides registry_default_overrides = 10;
  optional bool is_noto = 11;
  repeated string languages = 12;
  optional string primary_language = 13;
  repeated string display_name = 14;
  optional string primary_script = 15;
  optional string minisite_url = 16;
}
